pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	return resourceapply.ApplyService(c.ctx, c.kubeClient.CoreV1(), c.eventRecorder, required)
+}
+
+func (c *TargetConfigReconciler) manageServiceWebhook(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*v1.Service, bool, error) {
+	required := resourceread.ReadServiceV1OrDie(bindata.MustAsset("assets/lws-operator/service_webhook.yaml"))
+	required.Namespace = c.namespace
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	return resourceapply.ApplyService(c.ctx, c.kubeClient.CoreV1(), c.eventRecorder, required)
+}
+
+func (c *TargetConfigReconciler) manageServiceAccount(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*v1.ServiceAccount, bool, error) {
+	required := resourceread.ReadServiceAccountV1OrDie(bindata.MustAsset("assets/lws-operator/serviceaccount.yaml"))
+	required.Namespace = c.namespace
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	return resourceapply.ApplyServiceAccount(c.ctx, c.kubeClient.CoreV1(), c.eventRecorder, required)
+}
+
+func (c *TargetConfigReconciler) manageCustomResourceDefinition(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*apiextensionv1.CustomResourceDefinition, bool, error) {
+	required := resourceread.ReadCustomResourceDefinitionV1OrDie(bindata.MustAsset("assets/lws-operator/leaderworkerset.x-k8s.io_leaderworkersets.yaml"))
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	if required.Spec.Conversion != nil &&
+		required.Spec.Conversion.Webhook != nil &&
+		required.Spec.Conversion.Webhook.ClientConfig != nil &&
+		required.Spec.Conversion.Webhook.ClientConfig.Service != nil {
+		required.Spec.Conversion.Webhook.ClientConfig.Service.Namespace = c.namespace
+	}
+
+	return resourceapply.ApplyCustomResourceDefinitionV1(c.ctx, c.apiextensionClient.ApiextensionsV1(), c.eventRecorder, required)
```

Comment:
I had problems with the CABundle overwrites: I have solved this as follows: https://github.com/atiratree/jobset-operator/blob/9f91afbd99c7d1a5aef50ecce182e39f77e36b80/pkg/operator/target_config_reconciler.go#L138-L165

Might need more work once cert-manage is used if it does more modifications.

Reply:
Applied.